internal/topic/topicreaderinternal/stream_reconnector_test.go

```diff
@@ -295,11 +305,91 @@ func TestTopicReaderReconnectorStart(t *testing.T) {
 	_ = reconnector.CloseWithError(ctx, nil)
 }
 
+func TestTopicReaderReconnectorWaitInit(t *testing.T) {
+	t.Run("OK", func(t *testing.T) {
+		mc := gomock.NewController(t)
+		defer mc.Finish()
+
+		reconnector := &readerReconnector{
+			tracer:     &trace.Topic{},
+			initDoneCh: make(empty.Chan),
+		}
+		reconnector.initChannelsAndClock()
+
+		stream := NewMockbatchedStreamReader(mc)
+
+		reconnector.readerConnect = readerConnectFuncMock(readerConnectFuncAnswer{
+			callback: func(ctx context.Context) (batchedStreamReader, error) {
+				return stream, nil
+			},
+		})
+
+		reconnector.start()
+		for i := 1; i <= 5; i++ {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+			err := reconnector.WaitInit(ctx)
+			cancel()
+			require.NoError(t, err)
+		}
+	})
+
+	t.Run("contextDeadlineInProgress", func(t *testing.T) {
+		mc := gomock.NewController(t)
+		defer mc.Finish()
+
+		reconnector := &readerReconnector{
+			tracer: &trace.Topic{},
+		}
+		reconnector.initDoneCh = make(empty.Chan, 1)
+		reconnector.initChannelsAndClock()
+
+		stream := NewMockbatchedStreamReader(mc)
+
+		reconnector.readerConnect = readerConnectFuncMock(readerConnectFuncAnswer{
+			callback: func(ctx context.Context) (batchedStreamReader, error) {
+				return stream, nil
+			},
+		})
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
A second is very long period for pause, use some ned millisecond.

Or use any hook/poll for wait good time for cancel context from parallel goroutine.

The library has func SpinWaitCondition for simpler wait.


A lot of test with second pause will slow test process dramatically.